### PR TITLE
Enable SecretsManager.get to load and return bytes

### PR DESCRIPTION
**Pull request by ysysys3074**
# TL;DR
If the secrets are a base-64 encoded string, `SecretsManager.get` will raise exception. So add `encoded_mode` to the function here to fix and help to load and return bytes.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/core/context_manager.py [hunk 1]
```diff
@@ -349,7 +349,9 @@ def __getattr__(self, item: str) -> _GroupSecrets:
         """
         return self._GroupSecrets(item, self)
 
-    def get(self, group: str, key: Optional[str] = None, group_version: Optional[str] = None) -> str:
+    def get(
+        self, group: str, key: Optional[str] = None, group_version: Optional[str] = None, encode_mode: str = "r"
```
Comment (eapolinario) [inline flytekit/core/context_manager.py:353]: what if we make this an enum (containing only 2 options: "r" or "rb")?

Comment (ysysys3074) [reply flytekit/core/context_manager.py:353]: good idea, I will add in next commit

Comment (ByronHsu) [reply flytekit/core/context_manager.py:353]: i think using string is ok. It is consistent with what we pass to `with open(fpath, <here>)` . Enum seems a bit overkilled

Comment (eapolinario) [reply flytekit/core/context_manager.py:353]: Fair enough. At the very least put a description in the function docstring, @ysysys3074 ?

Comment (wild-endeavor) [reply flytekit/core/context_manager.py:353]: you can do a literal type if you want.

Comment (ysysys3074) [reply flytekit/core/context_manager.py]: @eapolinario added in new commit

**Review by eapolinario — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
